security/advancedtls/advancedtls.go

```diff
@@ -90,21 +104,31 @@ type ClientOptions struct {
 	// function every time asked to present certificates to the server when a new connection is
 	// established. This is known as peer certificate reloading.
 	GetClientCertificate func(*tls.CertificateRequestInfo) (*tls.Certificate, error)
-	// VerifyPeer is a custom server authorization checking after certificate signature check.
-	// If this is set, we will replace the hostname check with this customized authorization check.
-	// If this is nil, we fall back to typical hostname check.
+	// VerifyPeer is a custom verification check after certificate signature check.
+	// If this is set, we will perform this customized check after doing the normal check(s)
+	// indicated by setting VType.
 	VerifyPeer CustomVerificationFunc
 	// ServerNameOverride is for testing only. If set to a non-empty string,
 	// it will override the virtual host name of authority (e.g. :authority header field) in requests.
 	ServerNameOverride string
+	// RootCertificateOptions is REQUIRED to be correctly set on client side.
 	RootCertificateOptions
+	// If setting this field to CertAndHostVerification, we would perform both the certificate
+	// verification and hostname verification.
+	// If setting to CertVerification, we would only perform verification for the certificate sent from
+	// the server. Setting this field without proper custom verification check would leave the
+	// application susceptible to the MITM attack.
+	// If setting to SkipVerification, we would skip all the verification. Setting this field without
+	// proper custom verification check would leave the peer completely unauthenticated, and is
+	// highly discouraged unless you really have reasons to do so.
```

Comment:
These comments are mostly repeated on the `VType` field for `ServerOptions`. So, maybe it makes more sense to add this detailed documentation around the enum, and just refer to that from here and from `ServerOptions`.